cli/command_repository_upgrade.go

```diff
@@ -0,0 +1,240 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/internal/epoch"
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/content"
+)
+
+type commandRepositoryUpgrade struct {
+	forceRollback   bool
+	skip            bool
+	blockUntilDrain bool
+
+	// lock settings
+	advanceNoticeInterval  time.Duration
+	ioDrainTimeout         time.Duration
+	statusPollInterval     time.Duration
+	maxPermittedClockDrift time.Duration
+
+	svc advancedAppServices
+}
+
+func (c *commandRepositoryUpgrade) setup(svc advancedAppServices, parent commandParent) {
+	cmd := parent.Command("upgrade", "Upgrade repository format.")
+
+	// TODO:
+	// cmd.GetFlag("upgrade-owner-id").Required()
+	// cmd.("owner-id", "The upgrade lock unique owner ID").Required().StringVar(&c.ownerId)
+
+	cmd.Flag("advance-notice", "Advance notice for upgrade to allow enough time for other Kopia clients to notice the lock").DurationVar(&c.advanceNoticeInterval)
+	cmd.Flag("io-drain-timeout", "Max time it should take all other Kopia clients to drop repository connections").Default(repo.DefaultRepositoryBlobCacheDuration.String()).DurationVar(&c.ioDrainTimeout)
+	cmd.Flag("status-poll-interval", "An advisory polling interval to check for the status of upgrade").Default("60s").DurationVar(&c.statusPollInterval)
+	cmd.Flag("max-clock-drift", "Maximum tolerated drift on clocks between all Kopia clients").Default("5m").DurationVar(&c.maxPermittedClockDrift)
+	cmd.Flag("block-until-drain", "Drain all the clients but do not perform the upgrade").BoolVar(&c.blockUntilDrain)
+
+	// TODO(small): make rollback a sub-command instead of a flag
```

Comment:
This looks like it's going to be a small change (I may be wrong)–should we just include ti in this PR?

Reply:
This ended up becoming a larger change, will look into this in a separate PR.